### PR TITLE
Azure CI - Fix Windows Tests

### PR DESCRIPTION
**Pull request by Necktschnagge**
Let us have a look which files are showing up in build directory

### CODE REVIEW

**Comment by Necktschnagge** [comment]
fixes #35

**Comment by Necktschnagge** [comment]
@r0ost3r I moved your changes to a separate branch attached to this pull request.

**Comment by Necktschnagge** [comment]
Looking at the log fie, you see no release test exe is created. https://dev.azure.com/Necktschnagge/341439fd-7b7d-482a-8ae7-6ae03646a61d/_apis/build/builds/11195/logs/128

Maybe a pendant to ` -DCMAKE_BUILD_TYPE=Release` is needed when cmake builds the targets. Idk.

**Comment by Necktschnagge** [comment]
This might be the solution: https://stackoverflow.com/questions/7724569/debug-vs-release-in-cmake#comment120891316_7725055

**Comment by Necktschnagge** [comment]
This might be the solution: https://stackoverflow.com/questions/7724569/debug-vs-release-in-cmake#comment120891316_7725055 @r0ost3r

File: azure-pipelines.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ stages:
        - script: git submodule update --init --recursive
          displayName: "Init Git Submodules"
          condition: succeeded()
-       - script: cmake -S . -B ./build/ -DCMAKE_PREFIX_PATH="C:\Qt\5.15.2\msvc2019_64" -DUSE_QT6:BOOL=FALSE
+       - script: cmake -S . -B ./build/ -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="C:\Qt\5.15.2\msvc2019_64" -DUSE_QT6:BOOL=FALSE
```
Comment (Necktschnagge) [inline azure-pipelines.yml:71]: does it also work when you remove this flag?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -135,18 +135,24 @@ stages:
        - script: git submodule update --init --recursive
          displayName: "Init Git Submodules"
          condition: succeeded()
-       - script: cmake -S . -B ./build/ -DCMAKE_PREFIX_PATH="C:\Qt\6.5.1\msvc2019_64"
+       - script: cmake -S . -B ./build/ -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH="C:\Qt\6.5.1\msvc2019_64"
```
Comment (Necktschnagge) [inline azure-pipelines.yml:138]: does it also work when you remove this flag?

**Review by Necktschnagge — COMMENTED** [review]
Can you check this?

It's better to not carry unneeded flags.
If it works without, remove it.

File: azure-pipelines.yml (hunk 1 above)
Comment (r0ost3r) [reply azure-pipelines.yml]: yeah

File: azure-pipelines.yml (hunk 2 above)
Comment (r0ost3r) [reply azure-pipelines.yml]: yeah

**Comment by r0ost3r** [comment]
Suggest deactivation of Qt_5 mac_11 and Qt_6 mac_11 @Necktschnagge 

Warning: You are using macOS 11.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it

File: azure-pipelines.yml [hunk 3]
```diff
@@ -77,28 +78,32 @@ stages:
...
+         displayName: "CMake: Compile and Link (Release)"
          condition: succeeded()
        - script: ./build/test/TEST-$(project_name)
          displayName: "Run Tests on Linux / Mac OS"
          condition: and(succeeded(), ne( variables['Agent.OS'], 'Windows_NT'))
-       - script: .\build\test\Debug\TEST-$(project_name).exe
+       - script: .\build\test\Release\TEST-$(project_name).exe
          displayName: "Run Tests on Windows"
-         condition: and(false, succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
+         condition: and(true, succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
```
Comment (Necktschnagge) [inline azure-pipelines.yml:92]: remove `true, `

File: azure-pipelines.yml [hunk 4]
```diff
@@ -141,12 +146,21 @@ stages:
...
+       - script: tree /F .\build\test
+         displayName: "Search Tests on Windows"
+         condition: and(true, succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
+       - script: tree /F .\build
+         displayName: "Search Tests on Windows"
+         condition: and(true, succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
+       - script: .\build\test\Release\TEST-$(project_name).exe
          displayName: "Run Tests on Windows"
-         condition: and(false, succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
+         condition: and(true, succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
```
Comment (Necktschnagge) [inline azure-pipelines.yml:166]: same here

File: azure-pipelines.yml [hunk 5]
```diff
@@ -141,12 +146,21 @@ stages:
...
+         displayName: "CMake: Compile and Link (Debug)"
+         condition: succeeded()
+       - script: cmake --build ./build --config Release --verbose
+         displayName: "CMake: Compile and Link (Release)"
          condition: succeeded()
        - script: ./build/test/TEST-$(project_name)
          displayName: "Run Tests on Linux / Mac OS"
          condition: and(succeeded(), ne( variables['Agent.OS'], 'Windows_NT'))
-       - script: .\build\test\Debug\TEST-$(project_name).exe
+       - script: tree /F .\build\test
```
Comment (Necktschnagge) [inline azure-pipelines.yml:158]: remove this step

File: azure-pipelines.yml [hunk 6]
```diff
@@ -141,12 +146,21 @@ stages:
...
+         displayName: "Search Tests on Windows"
+         condition: and(true, succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
+       - script: tree /F .\build
```
Comment (Necktschnagge) [inline azure-pipelines.yml:161]: keep this step and also add this step to the Qt5 job.

File: azure-pipelines.yml [hunk 7]
```diff
@@ -141,12 +146,21 @@ stages:
...
+         displayName: "Search Tests on Windows"
```
Comment (Necktschnagge) [inline azure-pipelines.yml:162]: `"Search Tests on Windows"`  -> `"tree"`

File: azure-pipelines.yml (hunk 5 above)
Comment (Necktschnagge) [reply azure-pipelines.yml:164]: Die Zeile un die zwei folgenden sollen weg.

File: azure-pipelines.yml (hunk 7 above)
Comment (Necktschnagge) [reply azure-pipelines.yml:168]: Search Tests on Windows 
ist einfach nur ein blöder Titel

File: azure-pipelines.yml [hunk 8]
```diff
@@ -77,28 +78,35 @@ stages:
...
+         condition: succeeded()
+       - script: cmake --build ./build --config Release --verbose
+         displayName: "CMake: Compile and Link (Release)"
          condition: succeeded()
        - script: ./build/test/TEST-$(project_name)
          displayName: "Run Tests on Linux / Mac OS"
          condition: and(succeeded(), ne( variables['Agent.OS'], 'Windows_NT'))
-       - script: .\build\test\Debug\TEST-$(project_name).exe
+       - script: tree /F .\build
+         displayName: "View tree of 'build\' on Windows"
```
Comment (Necktschnagge) [inline azure-pipelines.yml:91]: you don't need escaping here...

Comment (Necktschnagge) [reply azure-pipelines.yml:91]: same below

Comment (r0ost3r) [reply azure-pipelines.yml]: The trailing (back)slash is generally used to mark a directory.

**Pull request closed without merging**